### PR TITLE
[REVIEW] Fix validity check for columns

### PR DESCRIPTION
**Pull request by kaatish**
In algorithms that do not support valid masks, GDF_VALIDITY_UNSUPPORTED
is now thrown if the valid pointer is not null and the null_count is not
0 for the columns involved.

### CODE REVIEW

File: libgdf/src/tests/quantiles/quant_tester.cu [hunk 1]
```diff
@@ -178,6 +182,37 @@ TEST_F(gdf_quantile, IntegerVector)
...
+  col_in.valid = d_valid.data().get();
+  col_in.null_count = 1;//Should cause the quantile calls to fail
+  col_in.dtype = GDF_INT32;
+
+  size_t n_qs = 5;
+  size_t n_methods = 5;
+
+  std::vector<VType> v_baseline_approx{-1, 0, 0, 1, 7};
+  std::vector<std::vector<double>> v_baseline_exact{
+    {-1, -1, 0, -0.5, -1},
```
Comment (harrism) [inline libgdf/src/tests/quantiles/quant_tester.cu]: Fix staggered indenting on following few lines.

**Comment by harrism** [comment]
@kaatish Looks like `join_test` is failing now...

File: libgdf/src/tests/groupby/groupby-test.cu [hunk 2]
```diff
@@ -116,15 +118,19 @@ struct GroupTest : public GdfTest {
...
-    auto deleter = [](gdf_column* col){col->size = 0; rmmFree(col->data, 0);};
+    auto deleter = [](gdf_column* col){col->size = 0; rmmFree(col->data, 0); rmmFree(col->valid, 0); };
     gdf_col_pointer the_column{new gdf_column, deleter};
 
     // Allocate device storage for gdf_column and copy contents from host_vector
     rmmAlloc(&(the_column->data), host_vector.size() * sizeof(col_type), 0);
     cudaMemcpy(the_column->data, host_vector.data(), host_vector.size() * sizeof(col_type), cudaMemcpyHostToDevice);
 
+    int valid_size = gdf_get_num_chars_bitmask(host_vector.size());
+    rmmAlloc((void**)&(the_column->valid), valid_size, 0);
```
Comment (harrism) [inline libgdf/src/tests/groupby/groupby-test.cu]: After latest RMM merge, you need to use RMM_ALLOC. Also, there is no error checking in this function.  Need to use RMM_TRY, but that requires this function to return an error.

Comment (harrism) [reply libgdf/src/tests/groupby/groupby-test.cu]: Perhaps error checking should be a follow up PR. I will file an issue. For now, just replace rmmAlloc with RMM_ALLOC.

File: libgdf/src/tests/groupby/groupby-test.cu [hunk 3]
```diff
@@ -116,15 +118,19 @@ struct GroupTest : public GdfTest {
 
     // Create a new instance of a gdf_column with a custom deleter that will free
     // the associated device memory when it eventually goes out of scope
-    auto deleter = [](gdf_column* col){col->size = 0; rmmFree(col->data, 0);};
+    auto deleter = [](gdf_column* col){col->size = 0; rmmFree(col->data, 0); rmmFree(col->valid, 0); };
```
Comment (harrism) [inline libgdf/src/tests/groupby/groupby-test.cu]: Need to use RMM_FREE

File: libgdf/src/tests/groupby/groupby-test.cu (hunk 2 above)
Comment (kaatish) [reply libgdf/src/tests/groupby/groupby-test.cu]: I have added EXPECT_EQ for error checking.

**Pull request merged**